import datetime
import sys
import traceback
import asyncio
from secrete import Chenal_id, Flag



async def sendErorsTelegram(bot,id_chenal=Chenal_id.LogsErroors, sec_start=5):
    time_apgrade = datetime.datetime.now()
    # –ü–æ–ª—É—á–µ–Ω–∏–µ –∏–Ω—Ñ–æ—Ä–º–∞—Ü–∏–∏ –æ–± –æ—à–∏–±–∫–µ
    exc_type, exc_value, exc_traceback = sys.exc_info()
    traceback_details = traceback.format_exception(exc_type, exc_value, exc_traceback)
    title_errors = traceback_details[0]
    file_str_errors = ''.join(traceback_details[1:-1])
    name_errors = traceback_details[-1]
    messageErrorTelegram = f"‚ùóÔ∏è–í—Ä–µ–º—è –∏ –¥–∞—Ç–∞:\n{time_apgrade}\n" \
                           f"üîπ–°—Ç–µ–∫ –≤—ã–∑–æ–≤–∞:\n{title_errors}\n" \
                           f"üìï–ì–¥–µ –ø—Ä–æ–∏–∑–æ—à–ª–∞ –æ—à–∏–±–∫–∞:\n{file_str_errors}\n" \
                           f"üëâ–ù–∞–∑–≤–∞–Ω–∏–µ –æ—à–∏–±–∫–∏:\n{name_errors}\n" \
                           f"üîÉ–ü–µ—Ä–µ–∑–∞–ø–ø—É—Å–∫ —á–µ—Ä–µ–∑: {sec_start}"
    if Flag.send_logErrors:
        await asyncio.sleep(2)
        await bot.send_message(id_chenal, messageErrorTelegram)
    else:
        print(messageErrorTelegram)